### ISSUE TITLE
microdnf fails if package is already installed

### ISSUE THREAD

**Issue by ignatenkobrain** [rpm-software-management/microdnf#7]
Description of problem:
I played a little bit with microdnf because I would like to use it in
my dockerfiles. I hit a little bug. The micro dnf fails to install package if a package is already installed.   

Version-Release number of selected component (if applicable):
```
sh$ rpm -q dnf microdnf libdnf
dnf-2.0.0-1.fc26.noarch
microdnf-1-1.fc26.x86_64
libdnf-0.7.0-0.6gitf9b798c.fc26.x86_64
```

How reproducible:
Deterministinc

Steps to Reproduce:
1. rpm -e lftp 
2. microdnf install lftp
3. microdnf install lftp
4. echo $?

Actual results:
```
sh# rpm -q lftp
package lftp is not installed

sh# microdnf install lftp
Transaction: 1 packages
lftp-4.7.4-1.fc26.x86_64 (rawhide)
Complete.

sh# echo $?
0

sh# microdnf install lftp
Transaction: (empty)
error: Transaction did not go to writing phase, but returned no error(0)
[root@11ca859098ab /]# echo $?
1
```

Expected results:
Return code should be 0 if package is already installed.
Otherwise it might break installation of packages from Dockerfile
after adding new packages to base image.

Additional info:
dnf does not fail in such situation.

```
sh# dnf install lftp
Last metadata expiration check: 0:14:21 ago on Thu Dec 22 13:04:44 2016 UTC.
Package lftp-4.7.4-1.fc26.x86_64 is already installed, skipping.
Dependencies resolved.
Nothing to do.
Complete!

sh# echo $?
0
```

**Comment by dmach** [comment]
This seems to be partially fixed:
* microdnf no longer fails
* microdnf should print: Package lftp-4.7.4-1.fc26.x86_64 is already installed, skipping.

**Comment by alexxa** [comment]
For me, `microdnf install dnf -y` works fine for the first time. Every next run it fails: 

```
[vagrant@fvagrant ~]$  docker exec mycontainer bash -l -c 'rpm -q dnf microdnf'
dnf-2.6.0-3.git.40.c0f7fc6.module_a7788b02.noarch
microdnf-2-2.module_39876f37.x86_64
[vagrant@fvagrant ~]$ docker exec mycontainer bash -l -c 'microdnf install dnf'
Downloading metadata...
Transaction: (empty)
Running transaction test...
error: Transaction did not go to writing phase, but returned no error(0)
```

**Comment by glennswest** [comment]
I found that the cache had to be cleared after each install

**Comment by satanb4** [comment]
This  issue is still being faced for when we develop using our docker build processes:-
`error: Error running transaction: package libcurl-7.29.0-57.el7_8.1.x86_64 (which is newer than libcurl-7.29.0-57.el7.x86_64) is already installed`

**Comment by kontura** [comment]
I am not sure when exactly this was fixed but at least `microdnf-3.4.0-2.fc32.x86_64` with `libdnf-0.45.0-3.fc32.x86_64` works:
```
$ microdnf install lftp
Package                                                                                                               Repository                                    Size
Installing:
 lftp-4.9.2-1.fc32.x86_64                                                                                             updates                                   965.4 kB
Transaction Summary:
 Installing:        1 packages
 Reinstalling:      0 packages
 Upgrading:         0 packages
 Removing:          0 packages
 Downgrading:       0 packages
Downloading packages...
Running transaction test...
Installing: lftp;4.9.2-1.fc32;x86_64;updates
Complete.
$ echo $?
0
$ microdnf install lftp
Nothing to do.
$ echo $?
0

```
